packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -960,6 +960,10 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
                     break;
                 }
+                case api.SummaryType.Attachment: {
```

Comment:
This, in principle, is same as SummaryType.Handle. I think the difference is that handles only support referencing IDs from previous snapshots (using path), vs. here we have hard ID in hands.
I feel like we either need to make it more generic (i.e. change Attachment to ByID or something generic), or maybe roll it into Handle case somehow (unlikely, given that it supports only paths).
@arinwt - what do you think?

Reply:
After reading tanvir's comments above and this one, I feel like I might be missing what the intent is..

But yes, summary handle must reference by path to a specific snapshot. i.e. `<summary handle>/path/to/blob`. So if these attachments/blobs are uploaded to storage directly, I don't know if we have a way to reference them (by our current summary contract types) without a new type to indicate it is using the ID.

Notably, the summary handles used to reference the IDs directly; this might be another pro to switching back to IDs in runtime contracts, leaving the handle conversion to the driver (slightly off-topic, but could be another solution).

Reply:
I mean we _could_ make it work without a new type, by having a special prefix or something to indicate it is using ID directly rather than path. `!id:<blob id>` or something like that.. but I don't really like that.

Reply:
I am a bit confused as well. I thought the primary intent is to make garbage collection easier for SPO. But as per my shallow understanding, we don't need a new "attachment" type to make that happen. Unless there are other reasons that I am not aware of.

So my main comment is, we should reconsider whether we need a completely new type? I am referring to all the attachment types in this PR (IAttachment, ISummaryAttachment, ISnapshotAttachment etc.)